File: front-end-project/src/services/quizAttemptService.js
import { get, post } from './api';

const SERVICE_RELATIVE_PATH = 'quiz-attempts';

export const startQuizAttempt = (quizId, studentId = null) => {
  return post(`${SERVICE_RELATIVE_PATH}/start`, { quizId, studentId });
};

export const submitStudentAnswer = (attemptId, questionId, submissionDTO) => {
  return post(`${SERVICE_RELATIVE_PATH}/${attemptId}/questions/${questionId}/submit-answer`, submissionDTO);
};


export const getQuizAttemptDetails = (attemptId) => {
  return get(`${SERVICE_RELATIVE_PATH}/${attemptId}`);
};

export const getQuizResults = (quizId) => {
  return get(`${SERVICE_RELATIVE_PATH}/quizzes/${quizId}/results`);
};

File: src/main/java/codefusion/softwareproject1/service/QuizAttemptService.java
package codefusion.softwareproject1.service;

import codefusion.softwareproject1.dto.AnswerSubmissionDTO;
import codefusion.softwareproject1.dto.QuizAttemptDTO;
import codefusion.softwareproject1.dto.QuizResultDTO;
import codefusion.softwareproject1.entity.StudentAnswer;



public interface QuizAttemptService {

    QuizAttemptDTO startQuizAttempt(Long quizId, Long studentId);

    StudentAnswer submitStudentAnswer(Long attemptId, Long questionId, AnswerSubmissionDTO submissionDTO, Long studentId);

    QuizResultDTO getQuizResults(Long quizId);
}
